src/disk.c

```diff
@@ -15,40 +15,63 @@ See the file LICENSE for details.
 
 #define DEFAULT_ATA_UNIT 0
 
-
-int disk_read(char *destination, int start_block_index, int offset,
-              int num_bytes) {
-    int blocks_needed = ((offset + num_bytes) / ATA_BLOCKSIZE) + 1;
-    if (((offset + num_bytes) % ATA_BLOCKSIZE) == 0) {
-        blocks_needed--;
+int disk_read(char *destination, int start_block_index, int offset, int num_bytes) {
+	int blocks_needed = ((offset + num_bytes) / ATA_BLOCKSIZE) + 1;
+	if(((offset + num_bytes) % ATA_BLOCKSIZE) == 0) {
```

Comment:
in other parts of the code, we have been doing this in the inverse (only add 1 if we need to, instead of removing it if we don't). Can we change this to be the same way, both for cleanliness and consistency